### ISSUE TITLE
jpg files difficulty

### ISSUE THREAD

**Issue by huvennpa** [JuliusCode/MP4MUSEUM#33]
dear Julius, 

I discovered your mp4 museum as almost the perfect solution for what I need in our educational work for nature conservation in the Forests of Brabant, nearby Brussels, Belgium. 'Almost', as I have difficulties in showing images. Most jpg files are refused, causing the loop to stop and show just an mp4museum prompt. I inderstood that the size of images should be limited, but even if I minimise them to less then 750 kB, most of the photographs can't be shown. I worked both on pi 3a+ and pi zero, and they both have similar problems. Though it looks like a pi zero is even more sensitive to *.jpg troubles than a 3a+. I screened your site and the GitHub site, but couldn't find this issues mentioned before? 

Thanks in advance four your advice

**Comment by JuliusCode** [comment]
Hi,

after doing some testing on a pi3b, i did get the same error with files larger than 2mb, but i‘ve got a usb key with 100+ files smaller than 1mb that runs without problems. 

did you try resizing or compressing your images to reduce the filesize? 

i tried a few things for a quick fix but lacking any error output to start from, i first need to find a way to debug running vlc inside of the script.

best, Julius.

**Comment by huvennpa** [comment]
Dear Julius,

I show you on the picture included the prompt I see on the moment the loop stops. That file in annex is, by the way, a compressed jpg file as I tried to use on the MP4 loop. I reduced jpg files with different online jpg compressors, looking if that might have been the cause of the trouble, but didn't find any difference between the compressors up to now. I tried to reduce the size of all pictures to less than 750 kB.

I found out in the mean time that there is a difference in behauviour between the pi zero and the pi 3a+. I've found pictures that are shown on the 3A but seem to block the zero, but also the other way around. Up to now, no .mp4 file caused the same difficulty.

I hope we can fix it, because furthermore mp4museum really is exactly what we need in our nature education efforts now!

Kind regards,
Patrick

Van: Julius ***@***.***>
Verzonden: woensdag 16 augustus 2023 10:17
Aan: JuliusCode/MP4MUSEUM ***@***.***>
CC: HUVENNE Patrick ***@***.***>; Author ***@***.***>
Onderwerp: Re: [JuliusCode/MP4MUSEUM] jpg files difficulty (Issue #33)


Hi,

after doing some testing on a pi3b, i did get the same error with files larger than 2mb, but i've got a usb key with 100+ files smaller than 1mb that runs without problems.

did you try resizing or compressing your images to reduce the filesize?

i tried a few things for a quick fix but lacking any error output to start from, i first need to find a way to debug running vlc inside of the script.

best, Julius.

-

**Comment by JuliusCode** [comment]
Hi Patrick,

unfortunately i did not receive your attachments. 

in the meantime, i have been finding out that the problem seems to be in vlc player. the same images that crashs mp4museum cause an exception with vlc player stand-alone.
will be looking into this but i cannot promise anything right now..

best, Julius

**Comment by huvennpa** [comment]
Dear Julius, 

the image of the prompt can be seen via this wetransfer link: 

https://we.tl/t-WlFRJq08IT

hope this helps to find the solution. We're supporting for you ;-)

**Comment by JuliusCode** [comment]
Dear Patrick,

if you could try the following:

use the „how to hack“ to enable write access, and make a change in mp4museum.py

then, open nano and add `—image-fps 1 ` 
to the vlc parameters in line 46, for example right before -q

on my pi 3b, this fixed the error. on zero, it just skips over large files.
however, i did not get issues with the same files on pi4, with the standard version.


best, Julius.

**Comment by JuliusCode** [comment]
i‘ve double checked, it is in there, but the way i see it on the webpage, it is hard to see that image-fps has two - in front of it, take care

**Comment by huvennpa** [comment]
Dear Julius,

We're almost there...I succeeded in finding line 46, I noticed the double - etc

Only problem left: when I use 'CTRL O' in the nano editor, or 'CTRL X' followed by 'Y' and 'ENTER' to save the changes, nano confirms it wrote 107 lines. But when I restart the raspberry pi 3b afterwards, the SD card didn't notice the changes I've made and the problem wasn't solved. If I re-open the nano editor then, I see that my changes are lost. What could be the reason that the SD card 'forgets' the correction?

Thanks in advance, Patrick
